concent_api/core/payments/report.py

```diff
@@ -0,0 +1,202 @@
+    """ Prefetches related PaymentInfo to PendingResponse model queryset. """
+    return queryset.select_related('payment_info')
+
+
+def get_related_payments_for_pair(clients_ethereum_addresses: list) -> Payments:
+    """ Get all payments between given requestor/provider pair. """
+    regular_payments = []  # type: List[str]
+    settlement_payments = []  # type: List[str]
+    forced_subtask_payments = []  # type: List[str]
+    additional_verification_payments_for_requestor = []  # type: List[str]
+    additional_verification_payments_for_provider = []  # type: List[str]
+    for clients_pair in clients_ethereum_addresses:
+        regular_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=clients_pair.requestor,
+            provider_eth_address=clients_pair.provider,
+            min_block_timestamp=get_oldest_subtask_creation_date(),
+            transaction_type=TransactionType.BATCH,
+        )
+
+        settlement_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=clients_pair.requestor,
+            provider_eth_address=clients_pair.provider,
+            min_block_timestamp=get_oldest_subtask_creation_date(),
+            transaction_type=TransactionType.SETTLEMENT,
+        )
+
+        forced_subtask_payments = service.get_list_of_payments(  # pylint: disable=no-value-for-parameter
+            requestor_eth_address=clients_pair.requestor,
+            provider_eth_address=clients_pair.provider,
+            min_block_timestamp=get_oldest_subtask_creation_date(),
+            transaction_type=TransactionType.FORCED_SUBTASK_PAYMENT,
+        )
+
+        additional_verification_payments_for_requestor = service.get_covered_additional_verification_costs(  # pylint: disable=no-value-for-parameter
+            client_eth_address=clients_pair.requestor,
+            payment_ts=get_oldest_subtask_creation_date(),
+        )
+        additional_verification_payments_for_provider = service.get_covered_additional_verification_costs(  # pylint: disable=no-value-for-parameter
+            client_eth_address=clients_pair.provider,
+            payment_ts=get_oldest_subtask_creation_date(),
+        )
+
+    return Payments(
+        regular_payments=regular_payments,
+        settlement_payments=settlement_payments,
+        forced_subtask_payments=forced_subtask_payments,
+        additional_verification_payments_for_requestor=additional_verification_payments_for_requestor,
+        additional_verification_payments_for_provider=additional_verification_payments_for_provider,
+    )
+
+
+def get_oldest_subtask_creation_date() -> int:
+    """
+    Return creation date as timestamp of oldest Subtask stored in database.
+    If there are no Subtasks in database, return current timestamp.
+    """
+    oldest_subtask = Subtask.objects.order_by('created_at').first()
```

Comment:
I think that @dybi has some important remarks regarding this function in https://github.com/golemfactory/concent/pull/1073.

Reply:
I can't find any comment attached to this line of code. But we can wait for him :)

Reply:
https://github.com/golemfactory/concent/pull/1073#discussion_r250523208

Reply:
I added that what @cameel and @dybi discussed about this function in previous PR.

Reply:
Then please change:
1. Function name.
2. Function comment.

Now the issue with different version of GM is also valid here.

Reply:
@rwrzesien what other name and comment are You suggesting? There's nothing about that in [#1073(comment)](https://github.com/golemfactory/concent/pull/1073#discussion_r250523208).

Reply:
Let me ask this way - does function name reflects what the function is actually returning? The same for comment.